libcorn/src/lua.rs

```diff
@@ -0,0 +1,51 @@
+use mlua::prelude::*;
+use crate::Value;
+use mlua::Value as OtherValue;
+impl<'lua> ToLua<'lua> for Value<'lua> {
+    fn to_lua(self, lua: &'lua Lua) -> LuaResult<LuaValue<'lua>> {
+        match self {
+            Value::Object(map) => {
+                let table = lua.create_table()?;
+                for (k, v) in map {
+                    table.set(k, v)?;
+                }
+                Ok(mlua::Value::Table(table))
+            }
+            Value::Array(vec) => {
+                let table = lua.create_table()?;
+                for (i, v) in vec.into_iter().enumerate() {
+                    table.set(i + 1, v)?;
+                }
+                Ok(mlua::Value::Table(table))
+            }
+            Value::String(s) => Ok(s.to_lua(lua)?),
+            Value::Integer(i) => Ok(i.to_lua(lua)?),
+            Value::Float(f) => Ok(f.to_lua(lua)?),
+            Value::Boolean(b) => Ok(b.to_lua(lua)?),
+            Value::Null(_) => Ok(LuaValue::Nil),
+        }
+    }
+}
```

Comment:
From looking at the [mlua docs](https://github.com/khvzak/mlua#serialization-serde-support), it looks like this might not actually be necessary. Instead, you might just be able to turn on the `serialize` feature flag and get the impl for free, since `Value` already derives `Serialize` .

If not, since every path is returning `Ok` you can drop those from the patterns and instead do something like:

```rs
let res = match { /* snip */ };
Ok(res)
```